torch/quantization/fx/quantize.py

```diff
@@ -123,6 +124,29 @@ def assert_and_get_unique_device(module):
     return device
 
 
+def preprocess_qconfig_dict(qconfig_dict):
+    """ Split the original qconfig_dict to
+    a flattened dict (global, object_type and module_name) and
+    a list of tuples of regex and qconfig (module_name_regex)
+    since order is important for regex matching (there might be conflicts and we
+    will take the first match)
+    and is not important for other qconfig types.
```

Comment:
thoughts about just making all of them match in an ordered fashion?  It could be simpler both to the user as well as in the implementation as everything will work the same way.

Reply:
looks like there is a cost to do that, which is making the qconfig lookup O(N) instead of O(1), where N is the number of entries of qconfig_dict. I feel this is an implementation detail, it's good enough for user to know the precedence between different types of configuration, and the matching within one type of configuration can still be considered as matched in order (unless user provides duplicate keys, we may check for this if needed).

Reply:
I'd be surprised if the performance of this block of code was a bottleneck.  For qconfig sizes we can reasonably expect to see (let's say a median size of 1 rule, and maybe p95 of 100 rules (just a guess)), would this matter?

Reply:
I don't think this will be a bottleneck, but it doesn't sound like a worthwhile trade off to me. since
1). it looks more natural to do dictionary lookup for the qconfigs where order does not matter
2). it does not affect user experience since this is an implementation detail
3). we are reusing propagate_qconfig_ from eager mode, which expects a dictionary, will need to update the matching there as well if we want to remove the flatten step.

Reply:
actually I think an alternative is convert them to ordered dict, and we can probably use the same data format for different qconfig types, although still different code to get the qconfig. does that sounds good?

Reply:
> 2). it does not affect user experience since this is an implementation detail

sg.  I don't feel too strongly

> actually I think an alternative is to convert them to ordered dict

in the API, or in the implementation?  In implementation would make sense.  For the API, that might make it more cumbersome for users.

Reply:
> in the API, or in the implementation? In implementation would make sense. For the API, that might make it more cumbersome for users.

in implementation.